Assets/SequenceSDK/WaaS/Tests/WaaSSessionManagementTests.cs

```diff
@@ -125,5 +125,57 @@ public async Task TestGetSessionAuthProof()
                 tcs.TrySetException(new Exception(error));
             });
         }
+
+        [Test]
+        public async Task GetIdTokenTest()
+        {
+            var tcs = new TaskCompletionSource<bool>();
+
+            
+                var login = SequenceLogin.GetInstance();
+
+                login.OnLoginFailed += (error, method, email, methods) =>
+                {
+                    tcs.TrySetResult(false);
+
+                    Assert.Fail("Login failed: "+error);
+                };
+
+                SequenceWallet.OnWalletCreated += async wallet =>
+                {
+                    try
+                    {
+                        wallet.OnIdTokenRetrieved += (idToken) =>
+                        {
+                            Assert.IsNotNull(idToken);
+                            Assert.IsFalse(string.IsNullOrWhiteSpace(idToken.IdToken));
+                            tcs.TrySetResult(true);
+                        };
+
+                        wallet.OnFailedToRetrieveIdToken += (error) =>
+                        {
+                            tcs.TrySetResult(false);
```

Comment:
Use tcs.TrySetException here instead (as we aren't checking the result of the task)